prow/cmd/sidecar/BUILD.bazel

```diff
@@ -15,10 +16,19 @@ go_library(
 )
 
 container_image(
-    name = "image",
+    name = "base-image",
     base = "@git-base//image",
     directory = "/",
-    files = [":sidecar"],
+    visibility = ["//visibility:private"],
+)
+
+go_image(
+    name = "image",
+    base = ":base-image",
+    embed = [":go_default_library"],
+    goarch = "amd64",
+    goos = "linux",
+    pure = "on",
```

Comment:
It appears this technique doesn't quite work as I'd hoped - I think previously, the idea with these 'decorated' containers is to include the built binary as, e.g. `/sidecar` (so that it can be referenced from the Prow job when it is created I'd guess?)

Doing this still does not place the binary at `/sidecar`, instead it can be found at `/app/prow/cmd/sidecar/image.binary.runfiles/__main__/prow/cmd/sidecar/linux_amd64_pure_stripped/image.binary`

I think we'll need to either create a symlink here, or otherwise drop the 'decorator-specific' parts of this pull request and follow up later.

Reply:
https://github.com/kubernetes/test-infra/issues/7947

Reply:
we could maybe use a symlink, or we'll need to update the container rules to support setting `directory`, which imho doesn't really make sense for the lang_image rules given the runfiles etc.